src/app/clusters/thread-border-router-management-server/thread-border-router-management-server.cpp

```diff
@@ -0,0 +1,384 @@
+            AddDatasetResponse(ctx, status, dataset);
+        });
+        break;
+    case Commands::SetActiveDatasetRequest::Id:
+        HandleCommand<Commands::SetActiveDatasetRequest::DecodableType>(ctxt, [this](HandlerContext & ctx, const auto & req) {
+            if (!mAsyncCommandHandle.Get())
+            {
+                mPath               = ctx.mRequestPath;
+                mAsyncCommandHandle = CommandHandler::Handle(&ctx.mCommandHandler);
+                ctx.mCommandHandler.FlushAcksRightAwayOnSlowCommand();
+                Status status = HandleSetActiveDatasetRequest(IsFailSafeArmed(ctx.mCommandHandler.GetAccessingFabricIndex()), req);
+                if (status != Status::Success)
+                {
+                    OnActivateDatasetComplete(mSetActiveDatasetSequenceNumber,
+                                              ChipError(ChipError::SdkPart::kIMGlobalStatus, to_underlying(status)));
+                }
+            }
+            else
+            {
+                ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Busy);
+            }
+        });
+        break;
+    case Commands::SetPendingDatasetRequest::Id:
+        HandleCommand<Commands::SetPendingDatasetRequest::DecodableType>(ctxt, [this](HandlerContext & ctx, const auto & req) {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, HandleSetPendingDatasetRequest(req));
+        });
+        break;
+    default:
+        break;
+    }
+}
+
+CHIP_ERROR ServerInstance::ReadFeatureMap(BitFlags<Feature> & featureMap)
+{
+    bool panChangeSupported;
+    ReturnErrorOnFailure(mDelegate->GetPanChangeSupported(panChangeSupported));
+    if (panChangeSupported)
+    {
+        featureMap.Set(Feature::kPANChange);
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::ReadBorderRouterName(MutableCharSpan & borderRouterName)
+{
+    ReturnErrorOnFailure(mDelegate->GetBorderRouterName(borderRouterName));
+    ReturnErrorCodeIf(borderRouterName.size() > kBorderRouterNameMaxLength, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::ReadBorderAgentID(MutableByteSpan & borderAgentId)
+{
+    ReturnErrorOnFailure(mDelegate->GetBorderAgentId(borderAgentId));
+    ReturnErrorCodeIf(borderAgentId.size() != kBorderAgentIdLength, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ServerInstance::ReadThreadVersion(uint16_t & threadVersion)
+{
+    return mDelegate->GetThreadVersion(threadVersion);
+}
+
+CHIP_ERROR ServerInstance::ReadInterfaceEnabled(bool & interfaceEnabled)
+{
+    return mDelegate->GetInterfaceEnabled(interfaceEnabled);
+}
+
+CHIP_ERROR ServerInstance::ReadActiveDatasetTimestamp(Optional<uint64_t> & activeDatasetTimestamp)
+{
+    uint64_t activeDatasetTimestampValue;
+    Thread::OperationalDataset activeDataset;
+    if (mDelegate->GetDataset(activeDataset, Delegate::DatasetType::kActive) == CHIP_NO_ERROR &&
```

Comment:
Please parenthesize the sub-expressions to the `&&`

Reply:
Done